File: src/gestorAplicacion/personas/Persona.java
package gestorAplicacion.personas;

import java.io.Serializable;
import java.util.*;
import gestorAplicacion.gestion.*;
import gestorAplicacion.personas.*;

public abstract class Persona implements Serializable {
	private int documento;
	private String nombre;
	private int edad;

	public Persona(int documento, String nombre, int edad) {
		super();
		this.documento = documento;
		this.nombre = nombre;
		this.edad = edad;
	}


	public abstract float calcularPromedio();
	public abstract Facultad getFacultad();
	public abstract LineasEnfasis getLineaEnfasis();

	
	public int getDocumento() {
		return documento;
	}

	public void setDocumento(int documento) {
		this.documento = documento;
	}

	public String getNombre() {
		return nombre;
	}

	public void setNombre(String nombre) {
		this.nombre = nombre;
	}

	public int getEdad() {
		return edad;
	}

	public void setEdad(int edad) {
		this.edad = edad;
	}

}

File: srcpy/gestorAplicacion/personas/persona.py
from abc import ABC, abstractmethod
from excepciones.documentoInvalido import DocumentoInvalido


class Persona(ABC):
    def __init__(self, documento, nombre, edad):
        self.documento = documento
        self.nombre = nombre
        self.edad = edad

    @classmethod
    def checkDocumento(cls, documento):
        from gestorAplicacion.personas.profesor import Profesor
        from gestorAplicacion.personas.estudiante import Estudiante

        vinculadosGeneral = (
            Profesor.getListaProfesores() + Estudiante.getListaEstudiantes()
        )
        documentosGeneral = [x.getDocumento() for x in vinculadosGeneral]
        if documento in documentosGeneral:
            raise DocumentoInvalido("El documento ingresado ya está registrado")

    @abstractmethod
    def calcularPromedio(self):
        pass

    def getFacultad(self):
        pass

    def getLineaEnfasis(self):
        pass

    def getDocumento(self):
        return self.documento

    def setDocumento(self, documento):
        self.documento = documento

    def getNombre(self):
        return self.nombre

    def setNombre(self, nombre):
        self.nombre = nombre

    def getEdad(self):
        return self.edad

    def setEdad(self, edad):
        self.edad = edad
